sunrise_sunset.py

```diff
@@ -1,150 +1,111 @@
-
-if __name__ == "__main__":
+        # Calculate decension, zenith and latitude
+        sinDec = 0.39782 * sin(rad(tl))
+        cosDec = cos(asin(sinDec))
+        
+        cosZen = cos(rad(self.zenith))
+        radLat = rad(self.latitude)
+        sinLat = sin(radLat)
+        cosLat = cos(radLat)
+
+        h = deg(acos( (cosZen - (sinDec * sinLat)) / (cosDec * cosLat) ))
+        h = (h if isSunset else 360 - h) / 15
 
+        mean_time = h + ra - (0.06571 * t) - 6.622
+        utc_time = (mean_time - lngHour) % 24
+        local_time = (utc_time + self.localOffset) % 24
+
+        hour, minute = divmod(local_time, 1)
+        minute, second = divmod(minute * 60, 1)
+
+        return date.replace(hour=int(hour), minute=int(minute), second=int(second*60))
+
+if __name__ == '__main__':
     import argparse
-    parser = argparse.ArgumentParser(description='Compute sunset and sunrise for a given day, location, and zenith value.')
-    parser.add_argument('latitude', metavar='LAT', type=float, help='the latitude')
-    parser.add_argument('longitude', metavar='LON', type=float, help='the longitude')
-    parser.add_argument('-d', '--date', help='a date (format: YYYY-MM-DD)')
-    parser.add_argument('-o', '--offset', type=int, help='offset from UTC')
-    args = parser.parse_args()
+    parser = argparse.ArgumentParser(
+        description='Compute sunset and sunrise for a given day, location, and zenith.')
+    parser.add_argument('latitude', metavar='LAT', type=float,
+                        help='latitude (degrees)')
+    parser.add_argument('longitude', metavar='LONG', type=float,
+                        help='longitude (degrees)')
+    parser.add_argument('-d', '--date',
+                        help='a date (YYYY-MM-DD)')
+    parser.add_argument('-o', '--offset', type=int, default=0,
+                        help='offset from UTC')
+    parser.add_argument('-z', '--zenith', type=int, default=CIVIL_ZENITH,
+                        help='zenith of sun')
 
-    # INPUTS
-    # Longitude and latitude
-    latitude = args.latitude
-    longitude = args.longitude
-    # Sun's zenith for sunrise/sunset
-    zenith = CIVIL_ZENITH
-    # Offset from UTC (GMT)
-    localOffset = args.offset if args.offset else 0
-    # date
-    date = datetime.datetime.strptime(args.date, '%Y-%m-%d') if args.date else datetime.datetime.now()
-
-    # do the computing
-    rise_obj = SunriseSunset(dt=date, latitude=latitude,
-                             longitude=longitude, localOffset=localOffset,
-                             zenith=zenith)
-    rise_time, set_time = rise_obj.calculate()
-
-    print("Latitude: ", latitude)
-    print("Longitude: ", longitude)
-    print("Date: ", date)
-    print("Offset: ", localOffset)
-    print("Sunrise", rise_time)
-    print("Sunset", set_time)
+    args = parser.parse_args()
+    lat, long = args.latitude, args.longitude
+    date = datetime.striptime(args.date, '%Y-%m-%d') if args.date else datetime.now()
```

Comment:
This error is encountered when you use the "-d" argument:
`AttributeError: type object 'datetime.datetime' has no attribute 'striptime'`